packages/@o3r/rules-engine/builders/rules-engine-extractor/helpers/rules-engine.extractor.interfaces.ts

```diff
@@ -1,18 +1,49 @@
 /** Supported types for a fact */
-export type FactSupportedTypes = 'string' | 'number' | 'date' | 'object' | 'boolean';
+export type FactSupportedTypes = 'string' | 'number' | 'date' | 'object' | 'boolean' | 'array';
 
-/** Fact as description in the metadata file */
-export interface MetadataFact {
+/** Base metadata fact */
+export interface BaseMetadataFact {
```

Comment:
if you don't need `BaseMetadataFact` outside this file I would advise not to export it.